controllers/progressiverollout_controller.go

```diff
@@ -92,11 +94,21 @@ func (r *ProgressiveRolloutReconciler) Reconcile(ctx context.Context, req ctrl.R
 		}
 
 		// Get the Applications to update
-		scheduledApps := scheduler.Scheduler(apps, stage)
+		scheduledApps := scheduler.Scheduler(outOfSyncApps, stage)
 
 		for _, s := range scheduledApps {
-			// TODO: add sync method here
 			r.Log.Info("syncing app", "app", s)
+
+			_, err := r.syncApp(s.Name)
```

Comment:
The left-hand side variable is the synced `*argov1alpha.Application`, when there's no error. Is there anything we want to do with it?

Reply:
I guess it's to have a lambda style approach where you can do things like `r.syncApps(s.Name).Status()` or something like this in a single command.
Don't think we need to do something in particular right now.

Reply:
We're actually doing something funny with that one. the `sync` from Argo takes an `ApplicationRequest` that we get from the `Name`. I guess the left hand side there is meant to tell you whether the selector worked. It could be useful down the line for us.